service/interpreter/persistence.go

```diff
@@ -127,17 +111,56 @@ func (am *PersistenceManager) LoadDataObjects(ctx UnifiedContext, loadingPolicy
 	if loadingType == "" || loadingType == iwfidl.ALL_WITHOUT_LOCKING {
 		return am.GetAllDataObjects()
 	} else if loadingType == iwfidl.PARTIAL_WITHOUT_LOCKING || loadingType == iwfidl.PARTIAL_WITH_EXCLUSIVE_LOCK {
-		res := am.GetDataObjectsByKey(service.GetDataObjectsQueryRequest{
-			Keys: partialLoadingKeys,
-		})
-		return res.DataObjects
+		return LoadDataAttributes(partialLoadingKeys, am.GetAllDataObjects())
 	} else if loadingType == iwfidl.NONE {
 		return []iwfidl.KeyValue{}
 	} else {
 		panic("not supported loading type " + loadingType)
 	}
 }
 
+func LoadSearchAttributes(loadingKeys []string, allSearchAttributes []iwfidl.SearchAttribute) []iwfidl.SearchAttribute {
```

Comment:
As our/golang convention, the internal usage of a method/function should be kept as "private". In golang, the any var/method/func start with lower case is package private `loadSearchAttributes`. 
Also there is already another method `LoadSearchAttributes`, you may want to rename it to `internalLoadSearchAttributes` to be more clear.

Reply:
Or maybe `loadSearchAttributesByRequestedKeys`